integration-tests/modules/__tests__/notification/admin/notification.spec.ts

```diff
@@ -26,102 +30,140 @@ medusaIntegrationTestRunner({
-        expect(notifications[0]).toEqual(
-          expect.objectContaining({
+          const notification2 = {
             to: "test@medusajs.com",
             channel: "log",
             template: "product-created",
-          })
-        )
-      })
-
-      it("should allow retrieving a notification", async () => {
-        const notification1 = {
-          to: "test@medusajs.com",
-          channel: "email",
-          template: "order-created",
-        } as CreateNotificationDTO
-
-        const notification2 = {
-          to: "test@medusajs.com",
-          channel: "log",
-          template: "product-created",
-        } as CreateNotificationDTO
-
-        const [first] = await service.create([notification1, notification2])
-
-        const notification = await service.retrieve(first.id)
-        expect(notification).toEqual(
-          expect.objectContaining({
+          } as CreateNotificationDTO
+
+          await service.create([notification1, notification2])
+
+          const notifications = await service.list({ channel: "log" })
+          expect(notifications).toHaveLength(1)
+          expect(notifications[0]).toEqual(
+            expect.objectContaining({
+              to: "test@medusajs.com",
+              channel: "log",
+              template: "product-created",
+            })
+          )
+        })
+
+        it("should allow retrieving a notification", async () => {
+          const notification1 = {
             to: "test@medusajs.com",
             channel: "email",
             template: "order-created",
+          } as CreateNotificationDTO
+
+          const notification2 = {
+            to: "test@medusajs.com",
+            channel: "log",
+            template: "product-created",
+          } as CreateNotificationDTO
+
+          const [first] = await service.create([notification1, notification2])
+
+          const notification = await service.retrieve(first.id)
+          expect(notification).toEqual(
+            expect.objectContaining({
+              to: "test@medusajs.com",
+              channel: "email",
+              template: "order-created",
+            })
+          )
+        })
+      })
+
+      describe.only("Configurable notification subscriber", () => {
+        let eventBus: IEventBusModuleService
+        beforeAll(async () => {
+          eventBus = getContainer().resolve(ModuleRegistrationName.EVENT_BUS)
+        })
+
+        it("should successfully sent a notification when an order is creaed (based on configuration)", async () => {
+          const logSpy = jest.spyOn(logger, "info")
+          eventBus.emit("order.created", {
+            data: {
+              order: {
+                id: "1234",
+                email: "test@medusajs.com",
+              },
+            },
           })
-        )
+
+          // TODO: Obviously a random sleep isn't great here, opening the PR to see if others have any ideas on how we can wait for the subscriber.
```

Comment:
To people reviewing, this is the only open point that I'd like your opinion on. The current implementation of the local event bus doesn't allow for easily waiting for all subscribers to finish, so I am curious to hear what you think would be a nice way to be able to do that in tests.

Reply:
for sake of testing, if you subscribe again to "order.created" and that resolves a promise reference, you can maybe just await for it.

```
let resolve
const waitPromise = new Promise((ok) => resolve = ok)

// ...subscribe(..., () => resolve())

await waitPromise
```

Reply:
@carlos-r-l-rodrigues this doesn't work (I've also tried it) because the notification subscriber is async and it takes a bit of time to complete due to the DB ops, so the second subscriber will finish before the first one does.

Reply:
In that case I'm fine with the sleep for now.
It doesn't make much sense to have a feature to await subscribers.
and everything else will involve changing the pubsub, like emmiting "local" events when jobs start/finish proccessing.
which by the way maybe can be the same thing used to handle graceful shutdown in this module.

Reply:
@carlos-r-l-rodrigues we came up with a pretty nice solution with @adrien2p , have a look when you can